sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -210,6 +210,37 @@ case class Sort(
   override def output: Seq[Attribute] = child.output
 }
 
+case class Range(
+    start: Long,
+    end: Long,
+    step: Long,
+    numSlices: Int) extends LeafNode {
+  require(step != 0, "step cannot be 0")
+  val numElements: BigInt = {
+    val safeStart = BigInt(start)
+    val safeEnd = BigInt(end)
+    if ((safeEnd - safeStart) % step == 0 || safeEnd > safeStart ^ step > 0) {
```

Comment:
I don't think this is the place where we should start speeding thing up :)... I'd rather have something which is easier to read.